### ISSUE TITLE
✅ 🚀 Feature Request: Table of Contents offset option

### ISSUE THREAD

**Issue by Jorigamix** [elementor/elementor#9947]
- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

**What problem is your feature request going to solve? Please describe.**
The new Table of Contents widget is pretty cool and useful. However clicking on the anchor links makes my page scroll to the anchor... which is hidden behind my fixed header.

**Describe the solution you'd like**
An offset option in the control pannel in px or % so the anchor link can scroll to header minus the offset setting.

**Describe alternatives you've considered**
I tried to solve this issue with css but couldn't find a solution working with any browser.

**Additional context**
![Feature request elementor](https://user-images.githubusercontent.com/58294966/71249602-db63dc80-231d-11ea-9ac7-4efa12753e1d.jpg)

**Comment by notechup** [comment]
Yes! +1 for this. I am trying to use the TOC widget but it scrolls down and cuts my Htags in the middle :/ It should make it possible to scroll to the top of the element, or offset by x pixels. I really can't use it without anny option to offset and hav not been able to find as olution to it yet.

**Comment by dimoteca** [comment]
+1

**Comment by coenlinq** [comment]
+1 very much in need for this tweak. Please follow up on this thread.

**Comment by imMaddy** [comment]
Add the below CSS code and adjust the "top" size as per your header height (+10 or 20px)

.elementor-menu-anchor {
    top: -75px;
    position: relative;
}

**Comment by imMaddy** [comment]
> * [x]  I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
> * [x]  The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )
> 
> **What problem is your feature request going to solve? Please describe.**
> The new Table of Contents widget is pretty cool and useful. However clicking on the anchor links makes my page scroll to the anchor... which is hidden behind my fixed header.
> 
> **Describe the solution you'd like**
> An offset option in the control pannel in px or % so the anchor link can scroll to header minus the offset setting.
> 
> **Describe alternatives you've considered**
> I tried to solve this issue with css but couldn't find a solution working with any browser.
> 
> **Additional context**
> ![Feature request elementor](https://user-images.githubusercontent.com/58294966/71249602-db63dc80-231d-11ea-9ac7-4efa12753e1d.jpg)

Add the below CSS code and adjust the "top" size as per your header height (+10 or 20px)

.elementor-menu-anchor {
top: -75px;
position: relative;
}

**Comment by bentalgad** [comment]
+1

**Comment by Pixelharmonist** [comment]
<3

**Comment by bachckm1** [comment]
Up

**Comment by bachckm1** [comment]
how to fix ?

**Comment by JochenPetri** [comment]
In case you wanna target a specific heading, just put "selector" before imMaddys code in that specific headers Custom CSS:

selector .elementor-menu-anchor {
top: -75px;
position: relative;
}

**Comment by HelloHeitz** [comment]
> Add the below CSS code and adjust the "top" size as per your header height (+10 or 20px)
> 
> .elementor-menu-anchor { top: -75px; position: relative; }

perfectly worked

**Comment by stefanomonteiro** [comment]
I cannot find this class (`elementor-menu-anchor`) in the HTML. Is anyone else having issue with this solution?

**Comment by HelloHeitz** [comment]
> I cannot find this class (`elementor-menu-anchor`) in the HTML. Is anyone else having issue with this solution?

See by Google dev tool,  on frontend

**Comment by stefanomonteiro** [comment]
I ctr-F the whole html and couldn't find it. Thye might have changed it on a later update. 

See more here https://forum.elementor.com/fixing-common-issues-33/toc-menu-table-of-content-in-combination-with-toggle-and-sticky-menu-doesnt-t-link-to-the-right-place-12191

**Comment by joehdevwp** [comment]
+1

**Comment by eakl** [comment]
+1

**Comment by wmacmill** [comment]
+1

**Comment by tbba** [comment]
+1

**Comment by PascalClaro** [comment]
+ 1

**Comment by viveknath3222** [comment]
This worked for me
.elementor-menu-anchor {
    position: absolute;
    top: -120px;
}

**Comment by nicholaszein** [comment]
Hi everyone!

### 📢 We're happy to announce that the issue you raised was resolved in Elementor Pro 3.25.0! 🥳 

This version introduced a new system for handling anchors and smooth-scroll, and included a new **`Anchor Offset`** control under the **`Sticky`** settings that allows to set a proper offset from the top of the page for anchor links when targeted.

You can learn more about it here:
https://elementor.com/help/sticky-headers/

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

> [!NOTE]
> If you think this is still an issue caused by Elementor, feel free to open a new 🐞 [Bug Report](https://github.com/elementor/elementor/issues/new?assignees=&labels=status%2Fawaiting_triage&template=bug.yml) with detailed information so we investigate your issue better.

Cheers 🥂
